cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
+	endif()
+
+	set_target_properties(
         "Boost::boost"
         PROPERTIES
-        INTERFACE_COMPILE_DEFINITIONS "BOOST_ALL_NO_LIB=1"
+        INTERFACE_COMPILE_DEFINITIONS ${compile_definitions}
```

Comment:
I think I know what problem is about this code. You have to quote the `${compile_definitions}`, otherwise if `compile_definitions` is a list the arguments after first will be treated as an arguments to the `set_target_properties`, not the value for `INTERFACE_COMPILE_DEFINITIONS`.

Here is the fix:
```cmake
set_target_properties(
    "Boost::boost"
    PROPERTIES
    INTERFACE_COMPILE_DEFINITIONS "${compile_definitions}"
)
```